src/test/java/io/cryostat/recordings/RecordingTargetHelperTest.java

```diff
@@ -806,6 +807,176 @@ public Future<Metadata> answer(InvocationOnMock invocation)
                                 Map.of("template.name", "Profiling", "template.type", "TARGET"))));
     }
 
+    void shouldReplaceExistingRecording() throws Exception {
+        String recordingName = "existingRecording";
+        String targetId = "fooTarget";
+        String duration = "5000ms";
+        String templateName = "Profiling";
+        TemplateType templateType = TemplateType.TARGET;
+        ConnectionDescriptor connectionDescriptor = new ConnectionDescriptor(targetId);
+        IRecordingDescriptor existingRecording = createDescriptor(recordingName);
+        IConstrainedMap<String> recordingOptions = Mockito.mock(IConstrainedMap.class);
+        Metadata metadata =
+                new Metadata(Map.of("template.name", "Profiling", "template.type", "TARGET"));
+
+        Mockito.when(targetConnectionManager.executeConnectedTask(Mockito.any(), Mockito.any()))
+                .thenAnswer(
+                        invocation -> {
+                            TargetConnectionManager.ConnectedTask task = invocation.getArgument(1);
+                            return task.execute(connection);
+                        });
+
+        Mockito.when(recordingOptions.get(Mockito.any())).thenReturn(recordingName, duration);
+
+        Mockito.when(connection.getService()).thenReturn(service);
+        List<IRecordingDescriptor> existingRecordings = List.of(existingRecording);
+        Mockito.when(service.getAvailableRecordings()).thenReturn(existingRecordings);
+
+        Mockito.when(service.start(Mockito.any(), Mockito.any())).thenReturn(existingRecording);
+
+        Mockito.when(existingRecording.getState()).thenReturn(RecordingState.STOPPED);
+
+        TemplateService templateService = Mockito.mock(TemplateService.class);
+        IConstrainedMap<EventOptionID> events = Mockito.mock(IConstrainedMap.class);
+        Mockito.when(connection.getTemplateService()).thenReturn(templateService);
+        Mockito.when(templateService.getEvents(Mockito.any(), Mockito.any()))
+                .thenReturn(Optional.of(events));
+
+        Mockito.when(
+                        recordingMetadataManager.setRecordingMetadata(
+                                Mockito.any(), Mockito.anyString(), Mockito.any(Metadata.class)))
+                .thenAnswer(
+                        invocation -> {
+                            return CompletableFuture.completedFuture(invocation.getArgument(2));
+                        });
+
+        Mockito.doNothing().when(notification).send();
+
+        recordingTargetHelper.startRecording(
+                false,
+                "always",
+                connectionDescriptor,
+                recordingOptions,
+                templateName,
+                templateType,
+                metadata,
+                false);
+
+        Mockito.verify(service).start(Mockito.any(), Mockito.any());
```

Comment:
I think there should also be a `verify(service).close(existingRecording)` or something like that, and it should verify that the `close` is done before the `start`, since that's what "restarting" a recording actually means in Cryostat terms.